### PR TITLE
Do not return broken or closed clients to the pool

### PR DESCRIPTION
**Pull request by brianc**
Building off of https://github.com/brianc/node-pg-pool/pull/119

This was slated to go into 8.0, but after working on this I'm not sure it's a breaking change...previously if you released a broken or ended client you'd basically put your app into a state where you'd crash or hit undefined behavior in the future.  Perhaps we just merge this into the existing 7.x branch so more folks can get the change sooner.  what you think @charmander?

Either way I think this closes the final thing mentioned in #2047 

The only thing really left for 8.0 is to get some proper tests in place for handling self-signed versus "real" certificates so we can check the code is doing the right thing in travis in all cases.  I need to reference the jdbc postgres repo and find how the configure all those different situations to test against them.  After that, I can write tests, make sure it works, and release 8.0.  Then the fun work of 9.0 begins. 🏃

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ env:
 node_js:
   - lts/dubnium
   - lts/erbium
-  - 13
+  - 13.6
```
Comment (charmander) [inline .travis.yml]: What’s this for?

File: packages/pg-pool/index.js [hunk 2]
```diff
@@ -256,7 +260,7 @@ class Pool extends EventEmitter {
 
     client.release = (err) => {
       if (released) {
-        throw new Error('Release called on client which has already been released to the pool.')
+        throwOnDoubleRelease()
```
Comment (charmander) [inline packages/pg-pool/index.js]: "

File: packages/pg-pool/test/releasing-clients.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    // make a pool w/ only 1 client
+    const pool = new Pool({ max: 1 })
+    expect(pool.totalCount).to.eql(0)
+    const client = await pool.connect()
+    expect(pool.totalCount).to.eql(1)
+    expect(pool.idleCount).to.eql(0)
+    // end the client gracefully (but you shouldn't do this with pooled clients)
+    client.end()
+
+    // try to return it to the bool
```
Comment (charmander) [inline packages/pg-pool/test/releasing-clients.js]:
```suggestion
    // try to return it to the pool
```

File: .travis.yml (hunk 1 above)
Comment (brianc) [reply .travis.yml]: the test to iterate a stream multiple times w/ async iterator stopped working in node 13.7....I've had occasional breaks before on single semver minor versions of non LTS node...if this doesn't resolve itself w/ 13.8 I'll have to track it down & see if it was something they intentionally changed w/ async iterators on streams.

Comment (brianc) [reply .travis.yml]: btw...found the issue:

https://github.com/nodejs/node/pull/31508

**Comment by brianc** [comment]
image


having a bit of a mental breakdown trying to rebuild the history in a way that cleanly merges into master.  I know it _can_ be done, but for the sake of my own sanity I'm just going to open a new PR that replaces this one, with the same commits, but targets `master` properly.

**Pull request closed without merging**